opm/simulators/flow/GenericCpGridVanguard.cpp

```diff
@@ -207,6 +208,33 @@ doLoadBalance_(const Dune::EdgeWeightMethod             edgeWeightsMethod,
                                  eclState1, parallelWells);
         }
 
+        // Add inactive wells to all ranks with connections (not solved, so OK even without distributed wells)
+        std::unordered_set<unsigned> cellOnRank;
+        const auto& global_cells = this->grid_->globalCell();
+        for (const auto cell : global_cells) cellOnRank.insert(cell);
+        const auto& comm = this->grid_->comm();
+        const auto nranks = comm.size();
+        const auto rank = comm.rank();
+        const auto inactive_well_names = schedule.getInactiveWellNamesAtEnd();
+        for (const auto& well_name : inactive_well_names) {
+            const auto& well = schedule.getWell(well_name, schedule.size()-1);
+            std::vector<int> well_on_rank(nranks, 0);
+            for (const auto& conn: well.getConnections()) {
+                if (cellOnRank.count(conn.global_index()) > 0) {
+                    well_on_rank[rank] = 1;
+                    break;
+                }
+            }
+            std::vector<int> well_on_rank_global(nranks, 0);
+            comm.max(&well_on_rank[0], nranks);
```

Comment:
```suggestion
            int well_on_rank = 0;
            for (const auto& conn: well.getConnections()) {
                if (cellOnRank.count(conn.global_index()) > 0) {
                    well_on_rank = 1;
                    break;
                }
            }
            std::vector<int> well_on_rank_global(nranks, 0);
            comm.allgather(&well_on_rank, 1, well_on_rank_global.data());
```
Your version is correct, too.

Now that I think of it, we could also do this in one communication step as those collective communications are rather expensive.
```c++
std::size_t num_wells = inactive_well_names.size();
std::vector<int> well_on_rank(num_wells, 0);
std::size_t well_idx = 0;
for (const auto& well_name : inactive_well_names) {
  const auto& well = schedule.getWell(well_name, schedule.size()-1);
  for (const auto& conn: well.getConnections()) {
      if (cellOnRank.count(conn.global_index()) > 0) {
          well_on_rank[well_idx] = 1;
          break;
      }
  }
  ++well_idx;
}
// values from rank i will be at indices i*num_wells, i*num_wells + 1, ..., (i+1) * num_wells -1
std::vector<int> global_well_on_rank(num_wells * nranks, 0);
comm.allgather(well_on_rank.data(), inactive_well_names.size(), well_on_rank_global.data());
well_idx = 0;
for (const auto& well_name : inactive_well_names) {
  for (int i=0; i<nranks; ++i) {
     if (well_on_rank[i*num_wells + well_idx]) {
        parallelWells.emplace_back(well_name, i);
     }
  }
  ++well_idx;
}

Reply:
Thanks, this is better. Will update!